drivers/npmx/npmx_shell.c

```diff
@@ -649,9 +671,35 @@ static int cmd_charger_trickle_set(const struct shell *shell, size_t argc, char
 	return 0;
 }
 
-static int cmd_die_temp_set(const struct shell *shell, size_t argc, char **argv,
-			    npmx_error_t (*func)(npmx_charger_t const *p_instance,
-						 int16_t temperature))
+static bool die_temp_set_helper(const struct shell *shell, npmx_charger_t *charger_instance,
+				int16_t temperature, int16_t *p_temperature,
+				npmx_error_t (*func_set)(npmx_charger_t const *p_instance,
+							 int16_t temperature),
+				npmx_error_t (*func_get)(npmx_charger_t const *p_instance,
+							 int16_t *temperature))
```

Comment:
Those arguments can be put in a struct, beside the shell, to decrease the number of passed arguments.

Reply:
As [here](https://github.com/NordicSemiconductor/npmx-zephyr/pull/110#discussion_r1386598731).